bash_it.sh

```diff
@@ -1,117 +1,129 @@
-if [ -z "$BASH_IT_THEME" ]; then
+if [[ -z "$BASH_IT_THEME" ]]; then
 	_log_warning "BASH_IT_THEME variable not initialized, please upgrade your bash-it version and reinstall it!"
 	export BASH_IT_THEME="$BASH_THEME"
 	unset BASH_THEME
 fi
 
-# support 'plumbing' metadata
-cite _about _param _example _group _author _version
-cite about-alias about-plugin about-completion
-
-# libraries, but skip appearance (themes) for now
-_log_debug "Loading libraries(except appearance)..."
-LIB="${BASH_IT}/lib/*.bash"
-APPEARANCE_LIB="${BASH_IT}/lib/appearance.bash"
-for _bash_it_config_file in $LIB; do
-	if [ "$_bash_it_config_file" != "$APPEARANCE_LIB" ]; then
-		filename=${_bash_it_config_file##*/}
-		filename=${filename%.bash}
-		BASH_IT_LOG_PREFIX="lib: ${filename}: "
-		_log_debug "Loading library file..."
-		# shellcheck disable=SC1090
-		source "$_bash_it_config_file"
-	fi
-done
+# Load the base libraries ...
+# Note: Ordering is important and intentional
 
-# Load vendors
-BASH_IT_LOG_PREFIX="vendor: "
-for _bash_it_vendor_init in "${BASH_IT}"/vendor/init.d/*.bash; do
-	_log_debug "Loading \"$(basename "${_bash_it_vendor_init}" .bash)\"..."
-	# shellcheck disable=SC1090
-	source "${_bash_it_vendor_init}"
-done
-unset _bash_it_vendor_init
+# "utilities" - internal functions meant for use within the bash-it codebase
+BASH_IT_LOG_PREFIX='lib: utilities: '
+_log_debug 'Loading library file ...'
+# shellcheck source=./lib/utilities.bash
+source "${BASH_IT}/lib/utilities.bash"
+
+# "helpers" - generic functions meant for generic use by end users,
+#             often wrappers of functions found in utilities
+BASH_IT_LOG_PREFIX='lib: helpers: '
+_log_debug 'Loading library file ...'
+# shellcheck source=./lib/helpers.bash
+source "${BASH_IT}/lib/helpers.bash"
+
+# "search" - allows end users to search for matching aliases, plugins and completions in bash-it
+BASH_IT_LOG_PREFIX='lib: search: '
+_log_debug 'Loading library file ...'
+# shellcheck source=./lib/search.bash
+source "${BASH_IT}/lib/search.bash"
+
+# Load aliases, completion, plugins ...
+BASH_IT_LOG_PREFIX='core: main: '
 
-BASH_IT_LOG_PREFIX="core: main: "
 # Load the global "enabled" directory
 # "family" param is empty so that files get sources in glob order
 # shellcheck source=./scripts/reloader.bash
 source "${BASH_IT}/scripts/reloader.bash"
 
 # Load enabled aliases, completion, plugins
-for file_type in "aliases" "plugins" "completion"; do
+for file_type in 'aliases' 'plugins' 'completion'; do
 	# shellcheck source=./scripts/reloader.bash
-	source "${BASH_IT}/scripts/reloader.bash" "skip" "$file_type"
+	source "${BASH_IT}/scripts/reloader.bash" 'skip' "$file_type"
 done
 
 # Load theme, if a theme was set
 if [[ -n "${BASH_IT_THEME}" ]]; then
 	_log_debug "Loading \"${BASH_IT_THEME}\" theme..."
+
 	# Load colors and helpers first so they can be used in base theme
-	BASH_IT_LOG_PREFIX="themes: colors: "
+	BASH_IT_LOG_PREFIX='themes: colors: '
+	_log_debug 'Loading theme file ...'
```

Comment:
this pattern is also duplicated (for themes) and can be extracted into a function